pkg/events/definition_group.go

```diff
@@ -123,6 +124,9 @@ func (d *DefinitionGroup) AddBatch(givenDefs map[ID]Definition) error {
 	defer d.mutex.Unlock()
 
 	for id, def := range givenDefs {
+		for i := range def.params {
+			def.params[i].Zero = parse.ArgZeroValueFromType(def.params[i].Type)
```

Comment:
Filling the respective zero value in the initialisation saves us of parsing further to discover the type of the nil arg.